components/net/connector.rs

```diff
@@ -50,9 +50,31 @@ impl NetworkConnector for HttpsConnector {
 
 pub type Connector = HttpsConnector;
 
-pub fn create_ssl_client(ca_file: &PathBuf) -> OpensslClient {
+pub fn create_ssl_client(certs: &str) -> OpensslClient {
+    let mut certs = certs.to_string();
+    // certs include multiple certificates. We could add all of them at once,
+    // but if any of them were already added, openssl would fail to insert all
+    // of them.
     let mut ssl_connector_builder = SslConnectorBuilder::new(SslMethod::tls()).unwrap();
-    ssl_connector_builder.set_ca_file(ca_file).expect("could not set CA file");
+    loop {
+        if let Some(index) = certs.rfind("-----BEGIN CERTIFICATE-----") {
+            let cert = certs.split_off(index);
```

Comment:
We could avoid allocating and copying here, by keeping `cert` as `&str` instead of `String`, and doing:

```rust
let (cert, rest) = certs.split_at(i);
certs = rest;
```